cmd/umoci/repack.go

```diff
@@ -236,5 +240,20 @@ func repack(ctx *cli.Context) error {
 	}
 
 	log.Infof("created new tag for image manifest: %s", tagName)
+
+	if ctx.Bool("refresh-bundle") {
+		newMtreeName := strings.Replace(newDescriptorPath.Descriptor().Digest.String(), "sha256:", "sha256_", 1)
+		if err := generateBundleManifest(newMtreeName, bundlePath, fsEval); err != nil {
+			return errors.Wrap(err, "write mtree metadata")
+		}
+		if err := os.Remove(mtreePath); err != nil {
+			return errors.Wrap(err, "remove old mtree metadata")
+		}
+		meta.From = newDescriptorPath
+		if err := WriteBundleMeta(bundlePath, meta); err != nil {
```

Comment:
I'm not sure how mtree storage works yet, so please ignore if this is a false alarm, but I think we want:

1. Generate new mtree.
2. Write bundle metadata pointing at new mtree.
3. Remove old mtree.

That way,we always have a working state, because dying with the old value unremoved seems better than dying with metadata pointing at a removed old value.

Reply:
That's exactly what's happening here; `generateBundleManifest` is doing your 1 and 2. `WriteBundleMeta` is different metadata. Blame Aleksa for the names ;-)